zypp-curl/parser/mediablocklist.cc

```diff
@@ -492,7 +492,7 @@ void MediaBlockList::reuseBlocks(FILE *wfp, const std::string& filename)
       // build the hashtable
       uint rsumHashMask = 0;
       {
-        int i = 16;
+        uint i = 16;
 
         /* Try hash size of 2^i; step down the value of i until we find a good size */
         while ((2 << (i - 1)) > nblks && i > 4)
```

Comment:
A bit late after merge, but did you compile it? 
The expression is still ` int` because the `2` is an 'int'.  `2U << ...` would have done the trick.

Reply:
I compiled it and I think the sign warning went away, but I am no longer sure. Thanks for your change fixing the two shift cases!